core/source/Http.mint

```diff
@@ -297,25 +315,72 @@ module Http {
         delete this._requests[#{uid}]
 
         resolve(#{Result::Err({
+          headers: `getResponseHeaders()`,
           type: Http.Error::Timeout,
           status: `xhr.status`,
           url: request.url
         })})
       })
 
-      xhr.addEventListener('load', (event) => {
+      xhr.addEventListener('load', async (event) => {
         delete this._requests[#{uid}]
 
+        let contentType = xhr.getResponseHeader("Content-Type");
+        let responseText = await xhr.response.text();
+        let body;
+
+        if (contentType.startsWith("text/html")) {
+          const object =
+            (new DOMParser()).parseFromString(responseText, "text/html");
+
+          const errorNode =
+            doc.querySelector("parsererror");
+
+          if (errorNode) {
+            body = #{Http.ResponseBody::Text(`responseText`)};
+          } else {
+            body = #{Http.ResponseBody::HTML(`object`)};
+          }
+        } else if (contentType.startsWith("application/xml")) {
+          const object =
+            (new DOMParser()).parseFromString(responseText, "application/xml");
+
+          const errorNode =
+            doc.querySelector("parsererror");
+
+          if (errorNode) {
+            body = #{Http.ResponseBody::Text(`responseText`)};
+          } else {
+            body = #{Http.ResponseBody::XML(`object`)};
+          }
+        } else if (contentType.startsWith("application/json")) {
+          try {
+            body = #{Http.ResponseBody::JSON(`JSON.parse(responseText)`)};
+          } catch (e) {
+            body = #{Http.ResponseBody::Text(`responseText`)};
+          }
+        } else if (contentType.startsWith("text/")) {
+          body = #{Http.ResponseBody::Text(`responseText`)};
+        }
+
+        if (!body) {
```

Comment:
It seems to me this branch is impossible to fulfil, since `body` is always set beforehand?

Reply:
The `body` is `null` to begin with https://github.com/mint-lang/mint/pull/588/files/9138b3d682ec1e763e9c3c22960d84dee6420cc9#diff-732cea90b2b18e39219ed170ca563d893caa3698fb7a515f8673748d4c473545R330 and there is no global `else` branch, so it can be `null` if we didn't match any types.

Reply:
Also, shouldn't it turn the response body into a `File` only for `application/octet-stream` content-type responses?

Reply:
Oh, there's no `else` branch, I missed that 👍

Reply:
Well images, PDFs, document etc. should be files IMO, also the text version is always available in `bodyString`.